src/main/java/org/springframework/samples/petclinic/rest/VisitRestController.java

```diff
@@ -78,12 +78,13 @@
 	public ResponseEntity<Visit> addVisit(@RequestBody @Valid Visit visit, BindingResult bindingResult, UriComponentsBuilder ucBuilder){
 		BindingErrorsResponse errors = new BindingErrorsResponse();
 		HttpHeaders headers = new HttpHeaders();
-		if(bindingResult.hasErrors() || (visit == null) || (visit.getPet() == null)){
+		if(bindingResult.hasErrors() || (visit == null) || (visit.getPet() == null) || visit.getVet() == null){
```

Comment:
**MEDIUM** No needs to check on NPE. Serializer will check Visit for pet and vet